tools/gen_ucd_tables.py

```diff
@@ -1,347 +0,0 @@
-#
-# See the COPYRIGHT file at the top-level directory of this distribution.
-#
-# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
-# http://www.apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT
-# or http://opensource.org/licenses/MIT>, at your option. This file may not be
-# copied, modified, or distributed except according to those terms.
-
-
-import fileinput
-import re
-import os
-import sys
-
-from os.path import join
-
-sys.path.append(join(os.path.dirname(__file__), "pylib"))
-
-import common
-import rustout
-import unicode_utils
-
-from common import path, memoize
-from unicode_utils import is_surrogate
-
-
-OUTPUT_DIRS = {
-    'UCD_NORMAL': path("unic/ucd/normal/src/tables"),
-}
-
-
-def data_path(name): return join(common.UCD_DATA_DIR, name)
-
-
-def test_data_path(name): return join(common.UCD_TEST_DATA_DIR, name)
-
-
-# == Version ==
-
-@memoize
-def get_unicode_version():
-    with open(data_path("ReadMe.txt")) as readme_file:
-        pattern = "for Version (\d+)\.(\d+)\.(\d+) of"
-        return re.search(pattern, readme_file.read()).groups()
-
-
-def emit_unicode_version(dir):
-    with open(join(dir, 'unicode_version.rsv'), "w") as version_file:
-        rustout.emit_value(
-            __file__,
-            version_file, get_unicode_version(),
-            print_fun=lambda x: "UnicodeVersion { major: %s, minor: %s, micro: %s }" % x)
-
-
-# == Shared ==
-
-@memoize
-def get_unicode_data():
-    unicode_data = {}
-    range_start = None
-    for line in fileinput.input(data_path("UnicodeData.txt")):
-        data = line.split(";")
-        if len(data) != 15:
-            continue
-        cp = int(data[0], 16)
-        if is_surrogate(cp):
-            continue
-        if range_start != None:
-            for i in xrange(range_start, cp):
-                unicode_data[i] = data
-            range_start = None
-        if data[1].endswith(", First>"):
-            range_start = cp
-            continue
-        unicode_data[cp] = data
-    return unicode_data
-
-
-# == Gen_Cat ==
-
-# Mapping taken from Table 12 from:
-# http://www.unicode.org/reports/tr44/#General_Category_Values
-EXPANDED_CATEGORIES = {
-    'Lu': ['LC', 'L'], 'Ll': ['LC', 'L'], 'Lt': ['LC', 'L'],
-    'Lm': ['L'], 'Lo': ['L'],
-    'Mn': ['M'], 'Mc': ['M'], 'Me': ['M'],
-    'Nd': ['N'], 'Nl': ['N'], 'No': ['No'],
-    'Pc': ['P'], 'Pd': ['P'], 'Ps': ['P'], 'Pe': ['P'],
-    'Pi': ['P'], 'Pf': ['P'], 'Po': ['P'],
-    'Sm': ['S'], 'Sc': ['S'], 'Sk': ['S'], 'So': ['S'],
-    'Zs': ['Z'], 'Zl': ['Z'], 'Zp': ['Z'],
-    'Cc': ['C'], 'Cf': ['C'], 'Cs': ['C'], 'Co': ['C'], 'Cn': ['C'],
-}
-
-
-# == Normal ==
-
-@memoize
-def get_normal_props():
-    interestingprops = ["Full_Composition_Exclusion"]
-
-    props = {}
-    re1 = re.compile("^ *([0-9A-F]+) *; *(\w+)")
-    re2 = re.compile("^ *([0-9A-F]+)\.\.([0-9A-F]+) *; *(\w+)")
-
-    for line in fileinput.input(data_path("DerivedNormalizationProps.txt")):
```

Comment:
The actual libraries (and tests) sort of are the tests for the generation code. I'm definitely adding some tests into `ucd-normal` to make sure the excluded compositions don't compose though!